### PR TITLE
Spec fix server entities

### PR DESCRIPTION
**Pull request by jshaughn**
@israel-hdez , @josejulio please review/merge.

### CODE REVIEW

File: spec/models/manageiq/providers/hawkular/inventory/parser/middleware_server_entities_spec.rb [hunk 1]
```diff
@@ -137,27 +139,23 @@ def parsed_deployment
     end
 
     it 'assigns enabled status to a deployment with "up" metric' do
-      metric_data.data.first['value'] = 'up'
+      puts "1=#{metric_data}"
```
Comment (josejulio) [inline spec/models/manageiq/providers/hawkular/inventory/parser/middleware_server_entities_spec.rb]: Please remove this `puts`.

**Review by josejulio — CHANGES_REQUESTED** [review]
Just a minor issue, could you please fix it?

File: app/models/manageiq/providers/hawkular/inventory/parser/middleware_server_entities.rb [hunk 2]
```diff
@@ -91,11 +91,14 @@ def parse_datasource(datasource, inventory_object)
...
-        'Enabled'
-      else
-        'Disabled'
+      begin
+        if availability.first['value'] && availability.first['value'][1] == '1'
+          'Enabled'
+        else
+          'Disabled'
+        end
+      rescue
```
Comment (lucasponce) [inline app/models/manageiq/providers/hawkular/inventory/parser/middleware_server_entities.rb]: Interesting, this also test the empty case.

**Comment by lucasponce** [comment]
LGTM
@israel-hdez if you give me rights on the israel-hdez repo I can help also with the merge tasks.

File: app/models/manageiq/providers/hawkular/inventory/parser/middleware_server_entities.rb (hunk 2 above)
Comment (jshaughn) [reply app/models/manageiq/providers/hawkular/inventory/parser/middleware_server_entities.rb]: Er, ** Don't merge this **, it has extra files involved.  I'll fix it up.

**Comment by jshaughn** [comment]
@josejulio @lucasponce , OK, I rebased and removed that leftover 'puts'.  I think it's good now.

**Comment by josejulio** [comment]
@israel-hdez Rubocob doesn't "fail" the build: see https://travis-ci.org/israel-hdez/manageiq-providers-hawkular/jobs/314811298#L1917-L1918

```bash
app/models/manageiq/providers/hawkular/inventory/parser/middleware_server_entities.rb:94 W: Redundant `begin` block detected.
app/models/manageiq/providers/hawkular/inventory/parser/middleware_server_entities.rb:100 W: Avoid rescuing without specifying an error class.
The command "if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then pronto run -c hawkular-1328; fi" exited with 0.
185.24s$ bundle exec rake
```

**Comment by jshaughn** [comment]
@josejulio I always forget rubocop, let me run that.  Is there a way to limit it only to the changed files on the branch?

**Comment by josejulio** [comment]
I use the miq-helper repo, there is a rubocop script that i run like this 
`bundle exec path/to/rubocop.rb --branch`
https://github.com/zeari/miq-helpers/blob/master/rubocop.rb

**Comment by jshaughn** [comment]
OK, rubocop should be happy

**Comment by israel-hdez** [comment]
I still saw a rubocop warning :P
https://travis-ci.org/israel-hdez/manageiq-providers-hawkular/jobs/314830823#L1914

**Comment by jshaughn** [comment]
@israel-hdez That rc issue does not show up for me locally.  Not sure why not:
```
> rubocop ./app/models/manageiq/providers/hawkular/inventory/parser/middleware_server_entities.rb

1 file inspected, no offenses detected
```

**Comment by israel-hdez** [comment]
@jshaughn ok, let me check... Perhaps it's a new warning in version 0.49.1 that wasn't in 0.47.1. This latter version is being used by the bot in the official repositories, so we should be safe ignoring that one.
